### PR TITLE
[WalletConnect] implement eth_signTypedData

### PR DESCRIPTION
**Pull request by eelanagaraj**
### Description
- implements `eth_signTypedData`

### Other changes
- adds minimal additional error handling which should now be propagated as a response

### Tested
- Manually via `use-contractkit` and by extending the `run-in-memory-client.ts` script in the `celo-monorepo` (in [this small PR](https://github.com/celo-org/celo-monorepo/pull/7951))

### Related issues
- Fixes celo-org/celo-monorepo#7514
### Backwards compatibility

cc @medhak1 as this changes some of the structure in the `acceptRequest` function in `../saga.ts`, assuming that will be relevant for `personal_decrypt` or the other json RPC methods you're implementing!

### CODE REVIEW

File: packages/mobile/src/walletConnect/saga.ts [hunk 1]
```diff
@@ -134,21 +134,36 @@ export function* acceptRequest({
...
+          break
+        case SupportedActions.eth_signTypedData:
+          yield call(unlockAccount, account)
+          result = yield call(wallet.signTypedData.bind(wallet), account, JSON.parse(params[1]))
+          break
+        default:
+          error = WalletConnectErrors.JSONRPC_REQUEST_METHOD_UNSUPPORTED
+      }
+    } catch (e) {
+      Logger.debug(TAG + '@acceptRequest error obtaining result: ', e.message)
```
Comment (AlexBHarley) [inline packages/mobile/src/walletConnect/saga.ts:154]: I'd prefer we assign the error `WalletConnectErrors.GENERIC` here but no biggy

**Review by AlexBHarley — COMMENTED** [review]
Looks good to me but let's wait for @jeanregisser to approve

File: packages/mobile/src/walletConnect/saga.ts (hunk 1 above)
Comment (eelanagaraj) [reply packages/mobile/src/walletConnect/saga.ts:154]: Did it this way to make sure error is for sure set (i.e. no weird case of neither error nor result being set), but maybe this is less readable. Can change to that way!

**Comment by AlexBHarley** [comment]
@eelanagaraj just added the wallet team to the reviewers

**Review by jeanregisser — APPROVED** [review]
Looks great! 👍

**Comment by Celoqa** [comment]
Hey @eelanagaraj @AlexBHarley I  have verified this issue using  Android internal build and Test Flight 1.15.0 & observed following:
**Observations:** 

- [x] **For Web:** User is able to connect Valora application & able to see connected apps in setting using use-contractkit.vercel.app.
- [x] Also it is observed that user is able get window to give permission from "use-contractkit demo". And user is able to get  "signTypedDataSucceded" message after giving permission by tapping on Allow button.
- [x] **For mWeb:** Alsi user is able to connect with same above steps by browsing  use-contractkit.vercel.app in web.
**Attachment:** [Web_signTypedData.wmv](https://drive.google.com/file/d/1Y6k3vyLgT9LZEEzckIdY2LbVRz5Hgg1Y/view?usp=sharing)
[mWeb_signTypedData.wmv](https://drive.google.com/file/d/1bkIphObNfuLL53-Fpy1etYmvkxVDqAP5/view?usp=sharing)
**Verified On Devices:** Redmi Note 8 (9.0), iPhone SE (14.2), Chrome Version 91.0.4472.77
Can you please let us know if we need to test anything else ? 
Thanks..!!

**Comment by eelanagaraj** [comment]
thanks @Celoqa! from my end that looks good

**Comment by Celoqa** [comment]
Always a pleasure @eelanagaraj.

**Pull request merged**